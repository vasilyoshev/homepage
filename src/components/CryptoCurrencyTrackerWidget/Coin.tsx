import React from 'react';
import { CoinProps } from 'interfaces';
import styles from './Coin.module.scss';

export const Coin: React.FC<CoinProps> = ({ name, image, symbol, marketcap, price, priceChange }) => {
  return (
    <div className={styles.coinContainer}>
      <div className={styles.coinRow}>
        <div className={styles.coin}>
          <img src={image} alt='crypto' />
          <h1>{name}</h1>
          <p className={styles.coinSymbol}>{symbol}</p> 
        </div>
        <div className={styles.coinData}>
          <p className={styles.coinPrice}>${price.toFixed(2)}</p>       
          {priceChange < 0 ? (
            <p className={styles.coinPercentRed}>{priceChange.toFixed(2)}%</p>
          ) : (
            <p className={styles.coinPercentGreen}>{priceChange.toFixed(2)}%</p>
          )
          }
          <p className={styles.coinMarketcap}>
               Mtk Cap: ${marketcap.toLocaleString()}
          </p>
        </div>
      </div>
    </div>
  );
};
