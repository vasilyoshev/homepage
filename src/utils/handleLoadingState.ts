import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
