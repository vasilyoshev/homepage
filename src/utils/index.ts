export * from './handleLoadingState';
export * from './testUtils';
