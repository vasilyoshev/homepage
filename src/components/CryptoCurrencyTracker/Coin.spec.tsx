import { render } from 'utils';
import { Coin } from './Coin';

describe('Coin', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<Coin 
      name='str' 
      image='str' 
      symbol='str' 
      marketcap={0} 
      price={0} 
      priceChange={0} 
      volume={0}/>);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
