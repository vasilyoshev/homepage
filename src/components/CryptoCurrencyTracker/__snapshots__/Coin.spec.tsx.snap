// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin renders successfully 1`] = `
<DocumentFragment>
  <div
    class="coinContainer"
  >
    <div
      class="coinRow"
    >
      <div
        class="coin"
      >
        <img
          alt="crypto"
          src="str"
        />
        <h1>
          str
        </h1>
        <p
          class="coinSymbol"
        >
          str
        </p>
      </div>
      <div
        class="coinData"
      >
        <p
          class="coinPrice"
        >
          $0
        </p>
        <p
          class="coinVolume"
        >
          $0
        </p>
        <p
          class="coinPercentGreen"
        >
          0.00%
        </p>
        <p
          class="coinMarketcap"
        >
          Mtk Cap: $0
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
